engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/function/Function.java

```diff
@@ -202,8 +203,21 @@ static Object doCall(
         Object[] arguments,
         @Cached InteropApplicationNode interopApplicationNode,
         @CachedLibrary("function") InteropLibrary thisLib) {
-      return interopApplicationNode.execute(
-          function, EnsoContext.get(thisLib).emptyState(), arguments);
+      try {
+        return interopApplicationNode.execute(
+            function, EnsoContext.get(thisLib).emptyState(), arguments);
+      } catch (StackOverflowError err) {
+        CompilerDirectives.transferToInterpreter();
+        var asserts = false;
```

Comment:
Yesterday I noticed @AdRiley had problems to analyze `StackOverflowError` to find out where it happens because `PolyglotException` wrapping the error **hides the stack over flow error stacktrace**. There doesn't seem to be any way to dig the stack from the `PolyglotException` without hacky reflection, but we can log the stack before it gets converted - log it at this moment:
```
	at org.enso.interpreter.runtime.callable.function.Function$Execute.doCall(Function.java:205)
	at org.enso.interpreter.runtime.callable.function.FunctionGen$InteropLibraryExports$Cached.executeAndSpecialize(FunctionGen.java:122)
	at org.enso.interpreter.runtime.callable.function.FunctionGen$InteropLibraryExports$Cached.execute(FunctionGen.java:108)
	at com.oracle.truffle.api.interop.InteropLibrary$Asserts.execute(InteropLibrary.java:3132)
	at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue$SharedExecuteNode.doDefault(PolyglotValueDispatch.java:4607)
	at com.oracle.truffle.polyglot.PolyglotValueDispatchFactory$InteropValueFactory$SharedExecuteNodeGen$Inlined.executeAndSpecialize(PolyglotValueDispatchFactory.java:9528)
	at com.oracle.truffle.polyglot.PolyglotValueDispatchFactory$InteropValueFactory$SharedExecuteNodeGen$Inlined.executeShared(PolyglotValueDispatchFactory.java:9480)
	at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue$ExecuteNoArgsNode.doDefault(PolyglotValueDispatch.java:4714)
	at com.oracle.truffle.polyglot.PolyglotValueDispatchFactory$InteropValueFactory$ExecuteNoArgsNodeGen.executeImpl(PolyglotValueDispatchFactory.java:9987)
	at com.oracle.truffle.polyglot.HostToGuestRootNode.execute(HostToGuestRootNode.java:124)
	at com.oracle.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:745)
	at com.oracle.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:669)
	at com.oracle.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:602)
	at com.oracle.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:586)
	at com.oracle.truffle.runtime.OptimizedRuntimeSupport.callProfiled(OptimizedRuntimeSupport.java:266)
	at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue.execute(PolyglotValueDispatch.java:2642)
	at org.graalvm.polyglot.Value.execute(Value.java:928)
	at org.enso.polyglot.Function.execute(Function.scala:17)
	at org.enso.runner.Main.runMain(Main.java:904)
	at org.enso.runner.Main.runPackage(Main.java:857)
	at org.enso.runner.Main.handleRun(Main.java:753)
	at org.enso.runner.Main.mainEntry(Main.java:1134)
	at org.enso.runner.Main.lambda$launch$22(Main.java:1477)
	at org.enso.runner.Main$$Lambda/0x0000710df011ca00.call
	at org.enso.runner.Main.withProfiling(Main.java:1224)
	at org.enso.runner.Main.launch(Main.java:1473)
	at org.enso.runner.Main.launch(Main.java:1425)
	at org.enso.runner.Main.main(Main.java:1040)
```
if we are running with _assertions enabled_ (like on CI), let's dump the stack with high severity. When running without asserts on, use `enso --log-level debug --run over.enso` to dump the stack.

Reply:
@hubertp, if there is a _more standard pattern for logging_ with stacktrace on CI and not logging the stacktrace in production by default, but only with bigger verbosity that doesn't involve `assert`, please advice what it is. I'll be glad to use it instead.

Reply:
The problem is that CI simply uses `enso` as a CLI, with no additional options. So cli used by regular users would behave the same way. I will see if I can add customize it further for CI via some env var.